spec/lib/cheers_leader_spec.rb

```diff
@@ -1,48 +1,118 @@
 require 'cheers_leader'
 
 describe CheerLeaderList do
-  before do
-    arr = [[2, "luan", 2, 2], [1, "tuan", 1, 1], [3, "mai", 2, 3]]  
+  before do 
+    leader_template = Struct.new(:rank, :name, :sent, :received)
```

Comment:
this should be a class name. you can just name it `Leader = Struct.new(...)`

Reply:
to be very explicit, I prefer that you create a different class like this using keyword arguments

```rb
class Leader
  attr_accessor :name, :rank, :sent, :received

  def initialize(name:, rank:, sent:, received:)
    self.name = name
    self.rank = rank
    self.sent = sent
    self.received = received
  end
end

Reply:
Interesting, Thanks for explain this. Will put this in mind